tests/test_generate.py

```diff
@@ -277,6 +293,41 @@ def test_generate(yaml_dir, tmpdir):
     t7 = html_blob.updated
     assert t6 != t7
 
+    # Upload new blob, build only latest, and verify only latest is updated.
+    only_latest = True
+    new_metadata = "docs.metadata.newer"
+    swap_file(yaml_dir, yaml_dir / "docs.metadata", yaml_dir / new_metadata)
+    latest_html_blob_name = "python-doc-pipeline-test-2.1.2.tar.gz"
+
+    upload_yaml(yaml_dir, test_bucket)
+    generate.build_all_docs(test_bucket, storage_client, only_latest)
+
+    # Verify old version HTML is not updated.
+    html_blob = bucket.get_blob(html_blob.name)
+    t8 = html_blob.updated
+    assert t7 == t8
+
+    # Verify latest version HTML is updated.
+    latest_html_blob = bucket.get_blob(latest_html_blob_name)
+    t1_latest = latest_html_blob.updated
+    assert t7 != t1_latest
+
+    # Force generation of latest Python docs
+    language = "python"
+    generate.build_language_docs(test_bucket, language, storage_client, only_latest)
+
+    # Verify old version HTML is not updated.
+    html_blob = bucket.get_blob(html_blob.name)
+    t9 = html_blob.updated
+    assert t8 == t9
+
+    # Verify latest version HTML is updated.
+    latest_html_blob = bucket.get_blob(latest_html_blob_name)
+    t2_latest = latest_html_blob.updated
+    assert t1_latest != t2_latest
+
+    swap_file(yaml_dir, yaml_dir / "docs.metadata", yaml_dir / new_metadata)
```

Comment:
Consider swapping, uploading, and swapping back in consecutive lines to make it easy to follow.